Sage/carbon
# fix(button): pass aria-hidden='true' to icon (FE-4477)

closes #4579

### Proposed behaviour



Have `Button` control the visibility of its icon to assistive technologies by passing down an `aria-hidden` prop - setting it to `true` if the _button has an icon_ and is _not icon-only_.

This is to ensure voice control tools don't deem the icon as a part of the button text, so the user can use voice commands on `Button` like normal, such as clicking on the button when its text is spoken.

### Current behaviour

The icon within `Button` is recognised as part of the button text by Voice Control on Mac in Safari. Therefore when the user speaks the button text as the element to target via a voice command, Voice Control does not recognise it.

### Checklist




#### QA


### Additional context

Voice Control on Mac and Safari were used in the original raised issue.

### Testing instructions

Open Storybook and test using Voice Control on `Button` with icon. Check that voice commands where the button text is spoken (e.g. "click submit" for clicking on a button with text "Submit") are recognised.
